routes/blog.ts

```diff
@@ -0,0 +1,79 @@
+export{};
+var express = require('express');
+const router= express.Router();
+const {Blog}=require('../models/blog')
+const auth=require('../middleware/auth');
+const mongoose=require('mongoose');
+const admin=require('../middleware/admin');
+const validateObjId=require('../middleware/validateObjId');
+
+router.get('/',async(req:any,res:any)=> {
+    const blogs=await Blog.find();
+    res.send(blogs);
+   
+})
+
+router.get('/:id',validateObjId,async(req:any,res:any)=>{
+ 
+    const blog=await Blog.findById(req.params.id);
+    if(!blog)res.status(400).send('Blog with given ID not found');
+    res.send(blog);
+})
+
+router.post('/',auth,async(req:any,res:any)=>{                            // POST
+    const { title,content, author,published} = req.body;
+    let blog=new Blog({title,content,author,published});
+    blog=await blog.save();
+    res.status(200).send(blog);
```

Comment:
In case of POST API res status should be 201

Reply:
Done